### PR TITLE
Fix duplicated classes + add enforcer rule to detect these automatically

### PR DESCRIPTION
**Pull request by psiroky**
* fix to make sure there is always only one (with excpetions) class
   with the particular FQN on classpath

### CODE REVIEW

**Comment by psiroky** [comment]
I removed some Errai deps excludes as they were recently done directly in Erra and are no longer needed in UF.

**Comment by csadilek** [comment]
+1

File: pom.xml [hunk 1]
```diff
@@ -97,6 +105,60 @@
...
+                           See https://github.com/errai/errai/blob/master/errai-javax-enterprise/README -->
+                      <ignoreClass>javax.interceptor.InvocationContext</ignoreClass>
+                      <ignoreClass>javax.inject.Provider</ignoreClass>
+                      <ignoreClass>javax.inject.Named</ignoreClass>
+                      <ignoreClass>javax.inject.Scope</ignoreClass>
+                      <ignoreClass>javax.inject.Qualifier</ignoreClass>
+                      <ignoreClass>javax.enterprise.*</ignoreClass>
+                      <!-- Classes from gwt-user duplicated in errai-uibinder -->
+                      <ignoreClass>com.google.gwt.uibinder.rebind.UiBinderWriter</ignoreClass>
+                      <ignoreClass>com.google.gwt.uibinder.rebind.UiBinderGenerator</ignoreClass>
```
Comment (csadilek) [inline pom.xml]: @psiroky the goods news is we deleted that module in Errai 4. So, as soon as we upgrade we can delete those.

**Comment by csadilek** [comment]
master: 885dd93b6c6d94d962cab9465683df14662bb517

**Pull request closed without merging**